internal/projection/process/event_enricher.go

```diff
@@ -105,16 +105,36 @@ func GetBaseEnrichedEvent(event *types.Event, cluster map[string]interface{}, ho
 
 	enrichedEvent.ID = uuid.NewSHA1(namespace, []byte(enrichedEvent.Message+enrichedEvent.EventTime)).String()
 
-	if versions, ok := event.Metadata["versions"]; ok {
-		if v, ok := versions.(map[string]interface{}); ok {
-			enrichedEvent.Versions = v
+	if versions := getVersionsFromMetadata(event); versions != nil {
+		if releaseTag := getReleaseTagFromMetadata(event, versions); releaseTag != nil {
+			enrichedEvent.ReleaseTag = *releaseTag
 		}
+		enrichedEvent.Versions = versions
 	}
+
 	enrichedEvent.Cluster = cluster
 	enrichedEvent.InfraEnvs = infraEnvs
 	return enrichedEvent, err
 }
 
+func getVersionsFromMetadata(event *types.Event) map[string]interface{} {
+	if versionsInterface, ok := event.Metadata["versions"]; ok {
+		if versions, ok := versionsInterface.(map[string]interface{}); ok {
+			return versions
+		}
+	}
+	return nil
+}
+
+func getReleaseTagFromMetadata(event *types.Event, versions map[string]interface{}) *string {
```

Comment:
Why version as a parameter? We most likely don't want to depend on versions being extracted.

Reply:
Same es the former comment, it resides inside outer `versions`

Reply:
I think it would be more readable if we extract it directly from metadata, see above comment